app/domain/authentication/authn_jwt/signing_key/signing_key_settings_builder.rb

```diff
@@ -0,0 +1,131 @@
+        end
+
+        private
+
+        def validate_signing_key_parameters
+          single_signing_key_source
+          cert_store_only_with_jwks_uri
+          public_keys_have_issuer
+        end
+
+        def single_signing_key_source
+          check_no_signing_keys_source
+          check_all_signing_keys_sources
+          check_jwks_provider_uri_pair
+          check_jwks_uri_public_keys_pair
+          check_public_keys_provider_uri_pair
+        end
+
+        def check_no_signing_keys_source
+          return unless !jwks_uri && !provider_uri && !public_keys
+
+          raise Errors::Authentication::AuthnJwt::InvalidSigningKeySettings,
+                "One of #{JWKS_URI_RESOURCE_NAME}, #{PUBLIC_KEYS_RESOURCE_NAME}, and #{PROVIDER_URI_RESOURCE_NAME} have to be defined"
+        end
+
+        def check_all_signing_keys_sources
+          return unless jwks_uri && public_keys &&  provider_uri
+
+          raise Errors::Authentication::AuthnJwt::InvalidSigningKeySettings,
+                "#{JWKS_URI_RESOURCE_NAME}, #{PUBLIC_KEYS_RESOURCE_NAME}, and #{PROVIDER_URI_RESOURCE_NAME} cannot be define simultaneously"
+        end
+
+        def check_jwks_provider_uri_pair
+          return unless jwks_uri && provider_uri
+
+          raise Errors::Authentication::AuthnJwt::InvalidSigningKeySettings,
+                "#{JWKS_URI_RESOURCE_NAME} and #{PROVIDER_URI_RESOURCE_NAME} cannot be define simultaneously"
+        end
+
+        def check_jwks_uri_public_keys_pair
+          return unless jwks_uri && public_keys
+
+          raise Errors::Authentication::AuthnJwt::InvalidSigningKeySettings,
+                "#{JWKS_URI_RESOURCE_NAME} and #{PUBLIC_KEYS_RESOURCE_NAME} cannot be define simultaneously"
+        end
+
+        def check_public_keys_provider_uri_pair
+          return unless public_keys && provider_uri
+
+          raise Errors::Authentication::AuthnJwt::InvalidSigningKeySettings,
+                "#{PUBLIC_KEYS_RESOURCE_NAME} and #{PROVIDER_URI_RESOURCE_NAME} cannot be define simultaneously"
+        end
+
+        def cert_store_only_with_jwks_uri
+          return unless ca_cert && !jwks_uri
+
+          raise Errors::Authentication::AuthnJwt::InvalidSigningKeySettings,
+                "#{CA_CERT_RESOURCE_NAME} can be define only with #{JWKS_URI_RESOURCE_NAME}"
+        end
+
+        def public_keys_have_issuer
+          return unless public_keys && !issuer
+
+          raise Errors::Authentication::AuthnJwt::InvalidSigningKeySettings,
+                "#{ISSUER_RESOURCE_NAME} is mandatory when #{PUBLIC_KEYS_RESOURCE_NAME} is set"
+        end
+
+        def signing_key_settings
+          @signing_key_settings_class.new(
+            uri: signing_key_settings_uri,
+            type: signing_key_settings_type,
+            cert_store: signing_key_settings_cert_store,
+            signing_keys: public_keys
+          )
+        end
+
+        def signing_key_settings_uri
+          return jwks_uri if jwks_uri
```

Comment:
If we in public-key case it just be nil?

Reply:
Yes